horovod/common/ops/gpu_operations.h

```diff
@@ -176,6 +176,22 @@ class GPUAllreduce : public AllreduceOp {
                                   double scale_factor,
                                   std::vector<TensorTableEntry>& entries);
 #endif
+#if HAVE_ROCM
+  void MemcpyInFusionBuffer(const std::vector<TensorTableEntry>& entries,
+                            const void*& fused_input_data, void*& buffer_data,
+                            size_t& buffer_len) override;
+
+  void MemcpyOutFusionBuffer(const void* buffer_data,
+                             std::vector<TensorTableEntry>& entries) override;
+
+  void ScaleMemcpyInFusionBuffer(const std::vector<TensorTableEntry>& entries,
+                                 const void*& fused_input_data,
+                                 void*& buffer_data, size_t& buffer_len,
+                                 double scale_factor);
+  void ScaleMemcpyOutFusionBuffer(void* buffer_data, size_t buffer_len,
+                                  double scale_factor,
+                                  std::vector<TensorTableEntry>& entries);
+#endif
```

Comment:
These declarations are identical for `HAVE_CUDA` and `HAVE_ROCM`. 

It would be better not to repeat them, but to list them just once behind a single preprocessor guard like:
```
#if HAVE_CUDA || HAVE_ROCM
...
#endif
```

Reply:
> > May I ask if the upstream has AMD GPUs available for CI?
> 
> I'm not sure if we would be able to set up a multi-GPU system with AMD GPUs on Buildkite here. An alternative might also be to integrate some CI job running on an appropriate external system if there is anything like that available on your end.
> 
> A good first step might also just be a job that builds Horovod in a container meant to be run on ROCM systems even if we don't actually run any actual tests there initially. That would catch many build time regressions at least.

Hi Max, we probably can set up an external CI for you. I will keep you posted on this issue.